Destination/StreamDestination.php

```diff
@@ -23,6 +32,53 @@ public function push($filename, LoggerInterface $logger)
     {
         $logger->info(sprintf('Copying %s to %s', $filename, $this->directory));
 
-        copy($filename, sprintf('%s/%s', $this->directory, basename($filename)));
+        copy($filename, ($destination = sprintf('%s/%s', $this->directory, basename($filename))));
+        touch($destination, filemtime($filename));
+
+        if (is_array($this->backups)) {
+            $this->backups[] = Backup::fromFile($destination);
+        } else {
+            $this->lazyLoadBackups();
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getIterator()
+    {
+        if (!is_array($this->backups)) {
+            $this->lazyLoadBackups();
+        }
+
+        return new ArrayIterator($this->backups);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function count()
+    {
+        if (!is_array($this->backups)) {
+            $this->lazyLoadBackups();
+        }
+
+        return count($this->backups);
+    }
+
+    /**
+     * Loads list of backups in lazy manner.
+     */
+    protected function lazyLoadBackups()
```

Comment:
Instead of having checks for is_array checks all over the place, just have:

```php
protected function getBackups()
{
    if (null !== $this->backups) {
        return $this->backups;
    }

    // ... contents of lazyloadbackups
}
```

That way you can just call `$this->getBackups()` any time you need them.